src/app/components/job-categories.tsx

```diff
@@ -1,12 +1,40 @@
+"use client";
 import type { NextPage } from "next";
+import { useEffect, useState } from "react";
+import { useInView } from "react-intersection-observer";
 import CompJoboffer from "./comp-joboffer";
+import { getActiveProposals } from "../actions/get-proposals";
 import styles from "./job-categories.module.css";
 
 export type JobCategoriesType = {
   className?: string;
+  initialData: any[];
 };
 
-const JobCategories: NextPage<JobCategoriesType> = ({ className = "" }) => {
+const PROPOSALS_TO_FETCH = 5; // Amount of proposals that would be loaded after scroll
+
+const JobCategories: NextPage<JobCategoriesType> = ({ initialData, className = "" }) => {
+  const [gigData, setGigData] = useState<any[]>(initialData); // further replace any[] to Gig[] Type from @types
+  const [offset, setOffset] = useState(PROPOSALS_TO_FETCH);
+  const [hasMore, setHasMore] = useState(true);
+  const [scrollTrigger, isView] = useInView();
+
+  const loadMoreProposals = async () => {
```

Comment:
please add comments defining what the function does

Reply:
First hook check if we run out with proposals from supabase
Second check if user scroll down page to load more proposals to frontend